### PR TITLE
Docs: First pass of README preparing for beta

### PR DESCRIPTION
**Pull request by jhaaaa**
- Just putting this here for now - preparing a README for the beta release.
- Based on discussion with Naomi, starting with the README from xmtp.js and we can pare down and adjust as necessary.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,31 +1,37 @@
...
 
-## Example app
+## Reference docs
 
-This repo provides an [example app](example) that you can use to experiment and explore implementation details.
+You can use the `xmtp-js` client SDK [reference documentation](https://xmtp-js.pages.dev/modules) as reference documentation for this SDK.
+
+## Quickstart app
+
+<!--Per our offsite discussion, I think this is a quickstart app, not an example app? If we make this change, I think we need to change the directory name.-->
```
Comment (jhaaaa) [inline README.md]: See comment

File: README.md [hunk 2]
```diff
@@ -55,17 +61,365 @@ npm i @xmtp/react-native-sdk
 npx pod-install
 ```
 
+<!--any thoughts on providing more details here?-->
```
Comment (jhaaaa) [inline README.md]: See comment

File: README.md [hunk 3]
```diff
@@ -55,17 +61,365 @@ npm i @xmtp/react-native-sdk
...
+
 We're working on testing the end-to-end installation and will provide more platform-specific configuration details.
 
 ### Configure for Android
 
 Your app must use Android `minSdkVersion = 22` to work with the `xmtp-react-native` SDK.
 
+<!--any thoughts on providing more details here?-->
```
Comment (jhaaaa) [inline README.md]: See comment

File: README.md [hunk 4]
```diff
@@ -55,17 +61,365 @@ npm i @xmtp/react-native-sdk
...
+
 We're working on testing the end-to-end installation and will provide more platform-specific configuration details.
 
-## Enable the example app to send push notifications
+## Usage
+
+The [XMTP message API](https://xmtp.org/docs/concepts/architectural-overview#network-layer) revolves around a network client that allows retrieving and sending messages to other network participants. A client must be connected to a wallet on startup. If this is the very first time the client is created, the client will generate a [key bundle](https://xmtp.org/docs/concepts/key-generation-and-usage) that is used to [encrypt and authenticate messages](https://xmtp.org/docs/concepts/invitation-and-message-encryption). The key bundle persists encrypted in the network using a [wallet signature](https://xmtp.org/docs/concepts/account-signatures). The public side of the key bundle is also regularly advertised on the network to allow parties to establish shared encryption keys. All this happens transparently, without requiring any additional code.
+
```
Comment (jhaaaa) [inline README.md:73]: See comments below

File: README.md [hunk 5]
```diff
@@ -55,17 +61,365 @@ npm i @xmtp/react-native-sdk
...
+for await (const message of await conversation.streamMessages()) {
+  console.log(`[${message.senderAddress}]: ${message.content}`)
+}
+```
+
+Currently, network nodes are configured to rate limit high-volume publishing from clients. A rate-limited client can expect to receive a 429 status code response from a node. Rate limits can change at any time in the interest of maintaining network health.
+
+## Create a client
+
+<!--Signer.ts doesn't exist in this repo. How might we update this text below?-->
```
Comment (jhaaaa) [inline README.md]: See comment

File: README.md [hunk 6]
```diff
@@ -55,17 +61,365 @@ npm i @xmtp/react-native-sdk
...
+
+## Handle different types of content
+
+All send functions support `SendOptions` as an optional parameter. The `contentType` option allows specifying different types of content than the default simple string standard content type, which is identified with content type identifier `ContentTypeText`.
+
+To learn more about content types, see [Content types with XMTP](https://xmtp.org/docs/concepts/content-types).
+
+Support for other types of content can be added by registering additional `ContentCodecs` with the `Client`. Every codec is associated with a content type identifier, `ContentTypeId`, which is used to signal to the client which codec should be used to process the content that is being sent or received.
+
+<!--not sure about the Codecs link - WDYT?-->
```
Comment (jhaaaa) [inline README.md:326]: See comment

File: README.md [hunk 7]
```diff
@@ -113,24 +467,67 @@ This example branch can serve as the basis for what you might want to provide fo
...
-6. Change the example app's environment to `production` in both places in `AuthView.tsx`.
+7. Change the quickstart app's environment to `production` in both places in `AuthView.tsx`.
 
-7. Replace `YOUR_FIREBASE_SENDER_ID` in the `PullController.ts` with your sender ID from Firebase.
+8. Replace `YOUR_FIREBASE_SENDER_ID` in the `PullController.ts` with your sender ID from Firebase.
 
-### Update the iOS example app to send push notifications
+### Update the iOS quickstart app to send push notifications
+
+<!--any thoughts on providing more details here?-->
```
Comment (jhaaaa) [inline README.md]: See comment

File: README.md (hunk 1 above)
Comment (nplasterer) [reply README.md]: @jhaaaa I talked again with @nakajima and I think on second thought these should actually stay as example. I don't think any word is super great but I think example is the most clear that it shouldn't be copy and pasted directly into their app and more is an example of what could be done. Maybe a note about database first application or a link to our RN playground for a more full featured reference could suffice.

File: README.md (hunk 2 above)
Comment (nplasterer) [reply README.md]: I guess we could add a comment about

```
In the ios directory, update your Podfile file as follows:
Set this value: platform :ios, '16.0'. This is required by XMTP.
Add this line: pod 'secp256k1.swift', :modular_headers => true. This is required for web3.swift.
```

File: README.md (hunk 3 above)
Comment (nplasterer) [reply README.md]: That's the only item that is needed for android specifically.

File: README.md [hunk 8]
```diff
@@ -55,17 +61,365 @@ npm i @xmtp/react-native-sdk
...
 
-## Enable the example app to send push notifications
+## Usage
+
+The [XMTP message API](https://xmtp.org/docs/concepts/architectural-overview#network-layer) revolves around a network client that allows retrieving and sending messages to other network participants. A client must be connected to a wallet on startup. If this is the very first time the client is created, the client will generate a [key bundle](https://xmtp.org/docs/concepts/key-generation-and-usage) that is used to [encrypt and authenticate messages](https://xmtp.org/docs/concepts/invitation-and-message-encryption). The key bundle persists encrypted in the network using a [wallet signature](https://xmtp.org/docs/concepts/account-signatures). The public side of the key bundle is also regularly advertised on the network to allow parties to establish shared encryption keys. All this happens transparently, without requiring any additional code.
+
+<!--as discussed, use the README from xmtp-js as the basis for this README. In the code samples, I changed @xmtp/xmtp-js in the imports to @xmtp/xmtp-react-native - correct?-->
```
Comment (nplasterer) [inline README.md]: Correct

File: README.md [hunk 9]
```diff
@@ -55,17 +61,365 @@ npm i @xmtp/react-native-sdk
...
+
+<!--I also changed the language in the code fences from tsx to js - correct?-->
```
Comment (nplasterer) [inline README.md]: It actually should be tsx.

File: README.md (hunk 5 above)
Comment (nplasterer) [reply README.md]: We use third web in our example but they could use whatever

```
import { ConnectWallet, useSigner } from "@thirdweb-dev/react-native";

const client = await XMTP.Client.create(useSigner(), "dev");

```

File: README.md [hunk 10]
```diff
@@ -55,17 +61,365 @@ npm i @xmtp/react-native-sdk
...
+| preCreateIdentityCallback | `undefined`                                                                       | `preCreateIdentityCallback` is a function that will be called immediately before a [Create Identity wallet signature](https://xmtp.org/docs/concepts/account-signatures#sign-to-create-an-xmtp-identity) is requested from the user.                                                                                                                                                                                                 |
+| preEnableIdentityCallback | `undefined`                                                                       | `preEnableIdentityCallback` is a function that will be called immediately before an [Enable Identity wallet signature](https://xmtp.org/docs/concepts/account-signatures#sign-to-enable-an-xmtp-identity) is requested from the user.                                                                                                                                                                                                |
```
Comment (nplasterer) [inline README.md]: We only support `env` currently

```suggestion

```

File: README.md [hunk 11]
```diff
@@ -55,17 +61,365 @@ npm i @xmtp/react-native-sdk
...
+When running in a browser, conversations are cached in `LocalStorage` by default. Running `client.conversations.list()` will update that cache and persist the results to the browsers `LocalStorage`. The data stored in `LocalStorage` is encrypted and signed using the Keystore's identity key so that attackers cannot read the sensitive contents or tamper with them.
+
+To disable this behavior, set the `persistConversations` client option to `false`.
+
+```js
+const clientWithNoCache = await Client.create(wallet, {
+  persistConversations: false,
+})
+```
+
```
Comment (nplasterer) [inline README.md]: Not supported

```suggestion

```

File: README.md (hunk 6 above)
Comment (nplasterer) [reply README.md:326]: We actually can use the content types defined in https://github.com/xmtp/xmtp-js-content-types

File: README.md (hunk 7 above)
Comment (nplasterer) [reply README.md]: On the branch iOS pushes should just work.

**Review by nplasterer — COMMENTED** [review]
I took a first pass on comments here but should take another pass on the code snippets.

**Comment by nplasterer** [comment]
I added a bunch of comments on here and then figured it would probably just be easiest if I made the changes myself so commit incoming.

**Comment by nplasterer** [comment]
Somehow this got in a weird git state so lets close this in favor of https://github.com/xmtp/xmtp-react-native/pull/76

**Pull request closed without merging**